### DISCUSSION TITLE
Octernships Query

### DISCUSSION THREAD

**Discussion by Amit-10101** [community/community#64768] [GitHub Education] [answered]
### Hi everyone,

Hi everyone,
I am a second-year student(3rd semester) and have some basic knowledge about coding. I can develop web pages using HTML, CSS & JavaScript and can do intermediate-level coding in Python, C, and C++. But I don't know much about DSA.

So, should I apply for Octernships now or first learn and practice, then apply for Octernships?

I will be thankful if anybody can answer this query of mine.

**Comment by arafajs** [comment, marked as answer]
@Amit-10101 
Based on your skills, you are eligible to apply for GitHub Octernships. However, it would be helpful if you had some knowledge of DSA (Data Structures and Algorithms) before applying. DSA is a fundamental concept in computer science and is essential for many software development tasks.

If you are not familiar with DSA, I recommend that you learn the basics first. There are many resources available online and in libraries. Once you have a good understanding of the basics, you can start practicing by solving problems on coding platforms like LeetCode or HackerRank.

Once you feel confident in your DSA skills, you can start applying for GitHub Octernships. Here are some tips for writing a strong application:

- Highlight your skills and experience in software development.
- Explain why you are interested in the Octernship program and what you hope to learn.
- Be specific about the projects you are interested in working on.
- Proofread your application carefully before submitting it.

**Reply by Amit-10101** [reply]
Thanks :)
